src/Components/Analyzers/test/ComponentParameterUsageAnalyzerTest.cs

```diff
@@ -0,0 +1,298 @@
+                        new DiagnosticResultLocation("Test0.cs", 11, 17)
+                    }
+                });
+        }
+
+        [Fact]
+        public void ComponentPropertyExpression_Ignores()
+        {
+            var test = $@"
+    namespace ConsoleApplication1
+    {{
+        using {typeof(ParameterAttribute).Namespace};
+        class TypeName
+        {{
+            void Method()
+            {{
+                System.IO.Console.WriteLine(new TestComponent().TestProperty);
+            }}
+        }}
+    }}" + ComponentTestSource;
+
+            VerifyCSharpDiagnostic(test);
+        }
+
+        [Fact]
+        public void ComponentPropertyExpressionInStatement_Ignores()
+        {
+            var test = $@"
+    namespace ConsoleApplication1
+    {{
+        using {typeof(ParameterAttribute).Namespace};
+        class TypeName
+        {{
+            void Method()
+            {{
+                var testComponent = new TestComponent();
+                for (var i = 0; i < testComponent.TestProperty.Length; i++)
+                {{
+                }}
+            }}
+        }}
+    }}" + ComponentTestSource;
+
+            VerifyCSharpDiagnostic(test);
+        }
+
+        [Fact]
+        public void RetrievalOfComponentPropertyValueInAssignment_Ignores()
+        {
+            var test = $@"
+    namespace ConsoleApplication1
+    {{
+        using {typeof(ParameterAttribute).Namespace};
+        class TypeName
+        {{
+            void Method()
+            {{
+                var testComponent = new TestComponent();
+                AnotherProperty = testComponent.TestProperty;
+            }}
+
+            public string AnotherProperty {{ get; set; }}
+        }}
+    }}" + ComponentTestSource;
+
+            VerifyCSharpDiagnostic(test);
+        }
+
+        [Fact]
+        public void ShadowedComponentPropertyAssignment_Ignores()
+        {
+            var test = $@"
+    namespace ConsoleApplication1
+    {{
+        using {typeof(ParameterAttribute).Namespace};
+        class TypeName
+        {{
+            void Method()
+            {{
+                var testComponent = new InheritedComponent();
+                testComponent.TestProperty = ""Hello World"";
+            }}
+        }}
+
+        class InheritedComponent : TestComponent
+        {{
+            public new string TestProperty {{ get; set; }}
+        }}
+    }}" + ComponentTestSource;
+
+            VerifyCSharpDiagnostic(test);
+        }
+
+        [Fact]
+        public void InheritedImplicitComponentPropertyAssignment_Ignores()
+        {
+            var test = $@"
+    namespace ConsoleApplication1
+    {{
+        using {typeof(ParameterAttribute).Namespace};
+        class TypeName : TestComponent
+        {{
+            void Method()
+            {{
+                this.TestProperty = ""Hello World"";
+            }}
+        }}
+    }}" + ComponentTestSource;
+
+            VerifyCSharpDiagnostic(test);
```

Comment:
Out of curiosity, why do the derived scenarios not generate a warning?

Reply:
Because it'd be similar to someone doing
```razor
@inherits MyBaseTypeWithAParameter

...

@code {
    public void OnClick()
    {
        MyProperty++;
    }
```